Automattic/wp-calypso
# Staging Sites: Clear state after sync and add success notification

Related to https://github.com/Automattic/dotcom-forge/issues/4137

## Proposed Changes

This pr adds the following changes:

- Refactor code to avoid duplicate logic
- Add success notification when sync is completed
- Clear state for selected direction and sync options after a successful sync.

## Testing Instructions
### Test 1: Clear options and toast notification
1. Create a WoA site. 
2. Add a staging site.
3. Initiate a sync.
4. Ensure that after the sync is completed, options are cleared and a toast with the message "'Site synced successfully." appears as shown:

![2023-10-09T15:45:46,703172634+03:00](https://github.com/Automattic/wp-calypso/assets/497103/9c9f616d-0d2b-41f1-a00b-d18e2c605e99)
5. Ensure that when you initiate a sync again, the toast disappears.

### Test 2: Ensure that functionality is as before
Due to refactor, ensure that everything is working as described on https://github.com/Automattic/wp-calypso/pull/82477



## Pre-merge Checklist
